cmd/fpga_admissionwebhook/README.md

```diff
@@ -1,97 +1,145 @@
-# Build and install Intel FPGA webhook for admission controller
+# Intel FPGA admission controller for Kubernetes
 
-### Dependencies
+# Table of Contents
 
-You must install and set up the following FPGA plugin modules for correct operation:
+* [Introduction](#introduction)
+* [Dependencies](#dependencies)
+* [Installation](#installation)
+    * [Getting the source code](#getting-the-source-code)
+    * [Deploying via the script](#deploying-via-the-script)
+        * [Pre-requisites](#pre-requisites)
+        * [Build the webhook image](#build-the-webhook-image)
+        * [Deploying via the script](#deploying-via-the-script)
+* [Mappings](#mappings)
+* [Next steps](#next-steps)
+
+# Introduction
+
+The FPGA admission controller is one of the components used to add support for Intel FPGA
+devices to Kubernetes.
+
+The FPGA admission controller webhook is responsible for performing mapping from user-friendly
+function IDs to the Interface ID and Bitstream ID that are required for FPGA programming by
+the [FPGA CRI-O hook](../fpga_crihook/README.md).
+
+Mappings are stored in namespaced custom resource definition (CRD) objects, therefore the admission
+controller also performs access control, determining which bitstream can be used for which namespace.
+More details can be found in the [Mappings](#mappings) section.
+
+The admission controller also keeps the user from bypassing namespaced mapping restrictions,
+by denying admission of any pods that are trying to use internal knowledge of InterfaceID or
+Bitstream ID environment variables used by the prestart hook.
+
+The admission controller can operate in two separate modes - preprogrammed or orchestration programmed.
```

Comment:
Perhaps worth noting that webhook's mode must be aligned with FPGA plugin's mode.